src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -1,165 +1,137 @@
     // Do not update anything if the cell sets are stil loading or if
     // the config does not exist yet.
-
     if (!config) {
       return;
     }
 
-    const spec = plots[selectedSpec].specGenerator(config);
-    generateData(spec);
-    setPlotSpec(spec);
-  }, [config]);
+    if (!cellSets.loading && !cellSets.error) {
+      switch (plots[selectedPlot].plotType) {
+        case 'embeddingPreviewMitochondrialReads':
+        case 'embeddingPreviewDoubletScore':
+          setPlot(<ContinuousEmbeddingPlot experimentId={experimentId} config={config} plotUuid={plots[selectedPlot].plotUuid} />);
+          break;
+
+        // Set this as default plot because default plot is 'sample' which is categorical
+        case 'embeddingPreviewBySample':
+        case 'embeddingPreviewByCellSets':
+        default:
+          setPlot(<CategoricalEmbeddingPlot experimentId={experimentId} config={config} plotUuid={plots[selectedPlot].plotUuid} />);
+          break;
+      }
+    }
+  }, [config, cellSets]);
 
   // If the user toggles to a different embedding, set the config to be the initial
   // state for that type of plot.
   useEffect(() => {
-    setConfig(plots[selectedSpec].initialConfig);
-  }, [selectedSpec]);
+    if (!isBrowser) return;
+    const { plotUuid, plotType } = plots[selectedPlot];
 
-  // Quick and dirty function to massage prepared data into a good shape.
-  // This will be changed once we actually load data from Redux.
-  /* eslint-disable no-param-reassign */
-  const generateData = (spec) => {
-    spec.data.forEach((s) => {
-      if (s.name === 'cellSets') {
-        s.values = [];
-
-        UMAP.forEach((cell, i) => {
-          s.values[cell.cluster_id] = {
-            name: `${cell.cluster_id}`,
-            cellSetId: cell.cluster_id,
-            cellIds: s.values[cell.cluster_id]?.cellIds ? [...s.values[cell.cluster_id].cellIds, i] : [i],
-            color: colorProvider.getColor(),
-          };
-        });
-      }
-
-      if (s.name === 'expression') {
-        s.values = { expression: UMAP.map((cell) => cell.doubletScore || 0) };
-      }
-
-      if (s.name === 'embedding') {
-        s.values = UMAP.map((cell) => [cell.UMAP_1, cell.UMAP_2]);
-      }
-    });
-  };
+    if (!config) {
+      dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    }
+  }, [selectedPlot]);
 
-  /* eslint-enable no-param-reassign */
   const updatePlotWithChanges = (obj) => {
-    const newConfig = _.cloneDeep(config);
-    _.merge(newConfig, obj);
-    setConfig(newConfig);
+    dispatch(updatePlotConfig(plots[selectedPlot].plotUuid, obj));
   };
 
   const renderPlot = () => {
-    if (error) {
+    if (selectedPlot === 'sample'
+      && !cellSets.loading
+      && filterCells(cellSets, config.selectedCellSet).length === 0) {
       return (
-        <PlatformError
-          description={error}
-          onClick={() => { }}
-        />
+        <Empty description='Your project has only one sample.' />
```

Comment:
I would rather handle that from the embedding categorical plot, since it is a check specific to it

Reply:
I don't really understand the condition. I derived this from one of Marcell's comment, which has the `selectedPlot === 'sample'` condition in it. If I were to move it inside `embeddingCategorical`, it would also be checked for the  2nd categorical plot (`coloredByCellSets`). Is that check relevant?

Reply:
I mean, it is not strictly speaking /relevant/, but we could put it there -- it can in theory happen with other configurations, such as plotting Scratchpad with no scratchpad cells.

Reply:
Not removed in latest commit.